task/net/README.md

```diff
@@ -1,6 +1,155 @@
 # About
 The `net` task implements a small netstack based on [_smoltcp_](https://github.com/smoltcp-rs/smoltcp)
 
+# Design of the SP Network Stack
+
+NOTE: This network stack is specialized to the SP's needs and is not, at the
+moment, intended as a fully general network stack.
+
+In particular, we currently only support IP and UDP, with a focus on IPv6.
+
+## The Net Task
+
+The `net` task is responsible for maintaining system network state. It includes:
+
+1. The Ethernet MAC and PHY drivers.
+2. Ethernet DMA support and buffer management.
+3. The IP stack itself (`smoltcp`).
```

Comment:
Extreme nit: are we styling this as `smoltcp` or _smoltcp_?  This section uses the former; previous text uses the latter.

Reply:
I would normally go for `smoltcp` since it's a crate name, but I'm flexible. I can edit it for consistency with what you had here.